packages/oas-utils/src/helpers/is-defined.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Type safe alternative to array.filter(Boolean)
+ *
+ * @example
+ *
+ * ```ts
+ * const dataArray = [1, null, 2, undefined, 3].filter(isDefined)
+ * ```
+ *
+ * @see https://jaketrent.com/post/typescript-type-safe-filter-boolean/
+ */
+export const isDefined = <T>(
+  value: T | null | undefined,
+): value is NonNullable<T> => value !== null && value !== undefined
```

Comment:
Should it not be 
```suggestion
): value is NonNullable<T> => value !== null && typeof value !== 'undefined'
```

?

Reply:
I think that would be the same thing?

Reply:
claude agreed lol

![image](https://github.com/user-attachments/assets/1e746055-5544-40eb-b173-2f699360b51c)